scripts/gas-refund-program/computeGasRefund.ts

```diff
@@ -13,64 +13,77 @@ import { GasRefundParticipation } from '../../src/models/GasRefundParticipation'
 import { init, resolveEpochCalcTimeInterval } from './common';
 import { EpochInfo } from '../../src/lib/epoch-info';
 import { CHAIN_ID_MAINNET } from '../../src/lib/constants';
+import Database from '../../src/database';
 
 const logger = global.LOGGER('GRP');
 
 async function startComputingGasRefundAllChains() {
-  await init({ epochPolling: true });
+  await init({
+    epochPolling: true,
+    dbTransactionNamespace: 'gas-refund-computation',
+  });
 
   const epochInfo = EpochInfo.getInstance(CHAIN_ID_MAINNET, true);
 
-  return Promise.allSettled(
-    GRP_SUPPORTED_CHAINS.map(async chainId => {
-      const lastEpochProcessed = await GasRefundParticipation.max<
-        number,
-        GasRefundParticipation
-      >('epoch', {
-        where: {
-          isCompleted: false,
-          chainId,
-        },
-      });
+  return Database.sequelize.transaction(async () => {
+    await Database.sequelize.query(
+      `LOCK TABLE "${GasRefundParticipation.tableName}" IN ACCESS EXCLUSIVE MODE;`,
```

Comment:
When does UNLOCK happen? When DB connection is closed?

Reply:
I'm using managed transaction and it handles all lock releasing on completion (both success or failure).